src/software/networking/threaded_network_listener.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <string>
+#include <thread>
+
+#include "software/networking/multicast_listener.h"
+
+template <class ListenerT, typename CallBackFunctionArgT>
+class ThreadedNetworkListener
```

Comment:
I think we should call this a `ThreadedMulticastListener` because we can't pass in non-multicast addresses

Reply:
Good point, changed.